src/palace.pyx

```diff
@@ -431,6 +434,15 @@ cdef class Context:
         """Update the context and all sources belonging to this context."""
         self.impl.update()
 
+    @staticmethod
+    def current_context() -> Optional[Context]:
+        """The context that is currently used."""
+        dev = Device(None)
+        current_impl = alure.Context.get_current()
+        dev.impl = current_impl.get_device()
+        current_ctx = Context(dev)
```

Comment:
The test is failing, I think because this create a new context instead of returning (oops you forgot to return it too) the old one.  Instead, you can create an empty `Context` then inject `alure::Context`, `alure::Device` and `alure::Listener` into it.  Message handler can be solved later.

Reply:
I have to return new context because I can't use a C++ object in a Python function, and so I think I should use `==` instead of `is`.

Reply:
> then inject [...] `alure::Listener` into it

Well, listener is a read-only attribute so I don't think we can do that.

Reply:
`readonly` applies only to Python (API) level.  Within Cython code it's still accessible.

Reply:
Referring to `Context(dev)`, please make `__init__` no-op, e.g. via the `None` idiom.